init_process.go

```diff
@@ -0,0 +1,406 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+// static file path
+const ConfigFileFolderPath = "init_process_fluent-bit_s3_config_files"
+const MainConfigFilePath = "init_process_fluent-bit.conf"
+const OriginalMainConfigFilePath = "/fluent-bit/etc/fluent-bit.conf"
+const InvokerFilePath = "invoker.sh"
+
+// default Fluent Bit command
+var FluentBitCommand = "exec /fluent-bit/bin/fluent-bit -e /fluent-bit/firehose.so -e /fluent-bit/cloudwatch.so -e /fluent-bit/kinesis.so"
+
+// global s3 client and flag
+var s3Client *s3.S3
+var exists3Client bool = false
+
+// global ecs metadata region
+var metadataReigon string = ""
+
+// set error log format
+func setErrorLog() {
+	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
+	log.SetPrefix("FluentBit Init Process -> ")
+}
+
+// create the invoker.sh
+// which will declare ECS Task Metadata as environment variables
+// and finally invoke Fluent Bit
+func createInvokerFile(filePath string) {
```

Comment:
I cannot see any difference for this function and `createMainConfigFile` in line 132. Both accept a file path and then create a file. Why do you make them separately?

Reply:
Thank you for your advice, have modified